generate_workspace/src/main/java/com/google/devtools/build/workspace/maven/DefaultModelResolver.java

```diff
@@ -182,12 +182,19 @@ public ModelSource resolveModel(Parent parent) throws UnresolvableModelException
   // For compatibility with older versions of ModelResolver which don't have this method,
   // don't add @Override.
   public void addRepository(Repository repository) {
-    repositories.add(repository);
+    addRepository(repository, false);
   }
 
   @Override
   public void addRepository(Repository repository, boolean replace) {
-    addRepository(repository);
+    if (repositories.contains(repository)) {
```

Comment:
Thank you. The `addRepositories(Repository repository, boolean replace)` is never called, so we skimped out on including the replace functionality. 

Assuming it's not particularly annoying, can you add a simple test case?

There isn't any `DefaultModelResolverTest` so you would need to create one, and mock the repositories.

Reply:
Done.

I also updated the repository store, since the previous implementation was relying on the internal implementation of [Repository.equals](https://maven.apache.org/ref/3.5.0/maven-model/apidocs/org/apache/maven/model/RepositoryBase.html#equals(java.lang.Object)), which is not documented as operating only on ID (even though it appears to be).

Reply:
Good catch on the equals.